encoding.go

```diff
@@ -111,9 +124,15 @@ func initEncoding(encodingName string) (*Encoding, error) {
 	}
 }
 
+func decodeVocab(blob []byte, vocab *map[string]int) error {
```

Comment:
+1

Reply:
I noticed in getEncoding() there is use of a mutex around access to `map[string]int`. This should use sync.Map (https://pkg.go.dev/sync#Map). Thoughts?

Reply:
From the link:

> The Map type is specialized. Most code should use a plain Go map instead, with separate locking or coordination, for better type safety and to make it easier to maintain other invariants along with the map content.

This code is already protected w/ the existing mutex you mentioned.  No need to change anything to maintain safety here, which is exactly what the Map type doc recommends.